scipy-stubs/signal/windows/_windows.pyi

```diff
@@ -29,36 +35,185 @@ __all__ = [
     "tukey",
 ]
 
-def general_cosine(M: Untyped, a: Untyped, sym: bool = True) -> Untyped: ...
-def boxcar(M: Untyped, sym: bool = True) -> Untyped: ...
-def triang(M: Untyped, sym: bool = True) -> Untyped: ...
-def parzen(M: Untyped, sym: bool = True) -> Untyped: ...
-def bohman(M: Untyped, sym: bool = True) -> Untyped: ...
-def blackman(M: Untyped, sym: bool = True) -> Untyped: ...
-def nuttall(M: Untyped, sym: bool = True) -> Untyped: ...
-def blackmanharris(M: Untyped, sym: bool = True) -> Untyped: ...
-def flattop(M: Untyped, sym: bool = True) -> Untyped: ...
-def bartlett(M: Untyped, sym: bool = True) -> Untyped: ...
-def hann(M: Untyped, sym: bool = True) -> Untyped: ...
-def tukey(M: Untyped, alpha: float = 0.5, sym: bool = True) -> Untyped: ...
-def barthann(M: Untyped, sym: bool = True) -> Untyped: ...
-def general_hamming(M: Untyped, alpha: Untyped, sym: bool = True) -> Untyped: ...
-def hamming(M: Untyped, sym: bool = True) -> Untyped: ...
-def kaiser(M: Untyped, beta: Untyped, sym: bool = True) -> Untyped: ...
-def kaiser_bessel_derived(M: Untyped, beta: Untyped, *, sym: bool = True) -> Untyped: ...
-def gaussian(M: Untyped, std: Untyped, sym: bool = True) -> Untyped: ...
-def general_gaussian(M: Untyped, p: Untyped, sig: Untyped, sym: bool = True) -> Untyped: ...
-def chebwin(M: Untyped, at: Untyped, sym: bool = True) -> Untyped: ...
-def cosine(M: Untyped, sym: bool = True) -> Untyped: ...
-def exponential(M: Untyped, center: Untyped | None = None, tau: float = 1.0, sym: bool = True) -> Untyped: ...
-def taylor(M: Untyped, nbar: int = 4, sll: int = 30, norm: bool = True, sym: bool = True) -> Untyped: ...
+_Array_f8_1d: TypeAlias = np.ndarray[tuple[int], np.dtype[np.float64]]
+_Array_f8_2d: TypeAlias = np.ndarray[tuple[int, int], np.dtype[np.float64]]
+_Weights: TypeAlias = (
+    Sequence[AnyReal]
+    | np.ndarray[tuple[int], np.dtype[np.floating[Any]]]
+    | np.ndarray[tuple[int], np.dtype[np.integer[Any]]]
+    | np.ndarray[tuple[int], np.dtype[np.bool_]]
+)
```

Comment:
As much as I am a fan of shape-typed arrays, I'm afraid that we can't require specific shapes for input (for now), because numpy has very little support for it.
So for example even simple things like `np.array([1])` currently return `np.ndarray[tuple[int, ...]]` (although I believe that I fixed that, but that'll be in the upcoming 2.2.0 release, and we currently have to support `numpy>=1.24`). 

tldr; for output it's perfectly fine to use shape-typing, but for input it's too restrictive at the moment.

Reply:
Okay got it I will change it.